### PR TITLE
[TwigBridge] Show Twig's loader paths on debug:twig command

### PR DESCRIPTION
**Pull request by yceruto**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

`bin/console debug:twig`:
twig-loader-paths

This information is not displayed anywhere ATM and it should be important to know:
 * The Twig's namespaces availables
 * The Twig's paths availables
 * The order that templates will be loaded ( regarding its namespace -> LOAD PRIORITY ! )

So it should help us to debug any issue related to circular templates reference, invalid namespaces, unsuccessful attempt to override a template, etc.

WDYT?

### CODE REVIEW

**Comment by yceruto** [comment]
I have another outputs for this list, which do you prefer?
 * **(A)**: `@namespace: path` (it looks more like how we use it) (current PR)
 * **(B)**: `path: namespace` (it looks more like how we define it in `twig.paths` configuration)
 * **(C)**: `path: @namespace` (it look like both A & B together)

This's **(B)**:
twig-loader-paths2

File: src/Symfony/Bridge/Twig/Command/DebugCommand.php [hunk 1]
```diff
@@ -145,9 +147,36 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $io->listing($items);
         }
 
+        $list = array();
+        foreach ($this->getLoaderPaths() as $namespace => $paths) {
+            $list = array_merge($list, array_map(function ($path) use ($namespace) {
+                return $namespace.($namespace ? ': ' : '').$path;
```
Comment (ogizanagi) [inline src/Symfony/Bridge/Twig/Command/DebugCommand.php]: What about substr the project dir if provided as a constructor argument of the command?

```php
if (null !== $this->projectDir && 0 === strpos($path, $this->projectDir)) {
    $path = ltrim(substr($file, strlen($this->projectDir)), DIRECTORY_SEPARATOR);
}
```

**Comment by ogizanagi** [comment]
For the record, I prefer the current version (A).

File: src/Symfony/Bridge/Twig/Command/DebugCommand.php [hunk 2]
```diff
@@ -145,9 +147,36 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+            }, $paths));
+        }
+        $io->section('Loader Paths');
+        $io->listing($list);
```
Comment (ogizanagi) [inline src/Symfony/Bridge/Twig/Command/DebugCommand.php]: Using a table might be a good alternative output to me.

**Comment by stof** [comment]
Outputting namespace first looks better to me (so A), as what is important is the different paths related to a given namespace. Paths used for separate namespaces are irrelevant regarding their order

File: src/Symfony/Bridge/Twig/Command/DebugCommand.php (hunk 2 above)
Comment (chalasr) [reply src/Symfony/Bridge/Twig/Command/DebugCommand.php]: Agree.

File: src/Symfony/Bridge/Twig/Command/DebugCommand.php (hunk 1 above)
Comment (yceruto) [reply src/Symfony/Bridge/Twig/Command/DebugCommand.php]: Nice idea! Done.

File: src/Symfony/Bridge/Twig/Command/DebugCommand.php (hunk 2 above)
Comment (yceruto) [reply src/Symfony/Bridge/Twig/Command/DebugCommand.php]: Done. (Btw, was necessary the `TableSeparator` and `*` to differentiate each row)

**Comment by yceruto** [comment]
Updated screenshot in description with the latest changes.

**Review by fabpot — APPROVED** [review]

**Comment by ogizanagi** [comment]
I find using `*` before each path and TableSeparator between each namespace really cluttering. :/

Which of the following output would you prefer?

1) <details><summary>Asterisk bullets, with separator (current)</summary>screenshot 2017-09-01 a 21 46 11</details>

2) <details><summary>No bullet, with separator </summary>screenshot 2017-09-01 a 21 46 31</details>

3) <details><summary>Asterisk bullets, no separator </summary>screenshot 2017-09-01 a 21 46 52</details>

4) <details><summary>Dash bullets, no separator </summary>screenshot 2017-09-01 a 21 47 07</details>

5) <details><summary>No bullet, no separator </summary>screenshot 2017-09-01 a 21 47 32</details>

My own vote would go to 4 or 5.

**Comment by yceruto** [comment]
and  6 :) (Dash bullets with blank separator):

**Comment by yceruto** [comment]
The issue that I see with no row separator is there is no grouping view by namespace, so one could get confused (missing namespace :confused:)

**Comment by yceruto** [comment]
My vote is for 7.

**Comment by ro0NL** [comment]
What about a `--layout=[1-7]` option :) And definitely support for layout=random :joy: 

7 is nice, agree with @yceruto; you getter a better view of path inheritance per namespace.

Tend to prefer `(Main) / (Root)` over `(None)` and have it on top.

**Comment by fabpot** [comment]
Thank you @yceruto.

**Pull request closed without merging**